lib/core/api.js

```diff
@@ -252,12 +256,22 @@ function postBlockToPlatform(data, groupId, blockIndex, callback) {
 
   api.log('postBlockToPlatform #' + blockIndex + ': using id ', groupId);
 
-  tidepool.uploadDeviceDataForUser(groupId, data, function(err, result) {
-    if (err) {
-      return sad(err, err);
-    }
-    return happy(result);
-  });
+  if(dataservices) {
+    tidepool.uploadDataServicesDataForUser(groupId, data, function(err, result) {
```

Comment:
It took me a while to read this correctly. Do I understand that `data` might be regular data or dataservices data and that `dataservices` in the function args is a boolean indicating which call to make? I'd vastly prefer a factory to produce a `postBlockToPlatform` function for each API, then just use each as appropriate. That way is still DRY but also w/o the if/else logic and confusing (& long) list of args.